### ISSUE TITLE
Training speed slow on V100

### ISSUE THREAD

**Issue by RockyXu66** [Skype-line/X-Avatar#19]
Hi, thanks for your excellent work.

I am attempting to retrain the model using your dataset. For training, I am using Nvidia Tesla V100. After setting up the environment, I made no changes to your original code and trained the **Clothed Human** model using the command `python train.py subject={Person_ID}_scan datamodule=XHumans_scan_smplx experiments=XHumans_occ_smplx datamodule.dataloader.dataset_path=/path/to/XHumans/Dataset/{Person_ID}`. The final evaluation metrics are the same as your pre-trained model. However, the training speed is ~6 s/it, which is slower than the 1.29 s/it mentioned in the supplement paper. The entire training process on my machine takes around 75 hours, which is three times longer than your reported training time of 24 hours (with batch_size=4, max_steps=45000).

My environment setting is Python 3.8.18, CUDA 11.1, pytorch 1.10.0, kaolin 0.12.0, pytorch-lightning 1.3.3, pytorch3d 0.7.4.

Additionally, I trained the SNARF project by the similar environment setting, which has a relatively  faster training speed as they mentioned (V100 is faster than 2080Ti).

Do you have any clues for the reason?

**Comment by MoyGcc** [comment]
Hi, thanks for your interest! Yeah, there was a misinterpretation in our supplementary material. The speed comparison was indeed conducted based on batch_size=1 and it's actually 1.29 seconds per each sample (>5 s/it, similar to your case). Sorry for the misunderstanding and we will revise it later.

**Comment by RockyXu66** [comment]
> Hi, thanks for your interest! Yeah, there was a misinterpretation in our supplementary material. The speed comparison was indeed conducted based on batch_size=1 and it's actually 1.29 seconds per each sample (>5 s/it, similar to your case). Sorry for the misunderstanding and we will revise it later.

Hi Chen, thanks for your reply. You mentioned "Training a model takes around 24h on a single Nvidia RTX 6000 GPU." in your supplement paper. Does that mean you stop earlier than 45,000 steps (with batch_size=4) when the model converges, or do you finish your training in 45,000 steps (with batch_size=1)?

**Comment by MoyGcc** [comment]
Hi Rocky, here 45000 steps are a relatively big number that allows training for at least 24 hours on clusters. In practice, we usually just train for 24 hours and haven't seriously tested whether the performance will increase a lot after training longer. Is there any visible difference from your side after you train longer? And also thank you for pointing that out!

**Comment by RockyXu66** [comment]
Here is the training log for scan 20 from WandB (https://api.wandb.ai/links/544200536/csfjyh7t). It was trained for a total of 77 hours across 104 epochs, which indicates that 24 hours of training will be around 30 epochs. According to the log, it seems like all loss values appear to be decreasing, with the exception of `bone_w_loss`.

**Comment by MoyGcc** [comment]
Thank you for the info! We will slightly update our config to obtain better performance.
